spec/jobs/amend_backorder_job_spec.rb

```diff
@@ -86,13 +86,33 @@
       expect(backorder.lines[0].quantity).to eq 1 # beans
       expect(backorder.lines[1].quantity).to eq 5 # chia
 
-      # We cancel the only order and that should reduce the order lines to 0.
+      # We increase quantities which should be reflected in the backorder:
+      beans.on_hand = -1
+      chia_item.quantity += 3
+      chia_item.save!
+
+      expect { subject.amend_backorder(order) }
+        .to change { beans.on_hand }.from(-1).to(11)
+        .and change { backorder.lines[0].quantity }.from(1).to(2)
+        .and change { backorder.lines[1].quantity }.from(5).to(8)
+
+      # We cancel the only order.
       expect { order.cancel! }
-        .to change { beans.reload.on_hand }.from(9).to(15)
-        .and change { chia_seed.reload.on_hand }.from(7).to(12)
+        .to change { beans.reload.on_hand }.from(11).to(17)
+        .and change { chia_seed.reload.on_hand }.from(4).to(12)
 
+      # But we decreased the stock of beans outside of orders above.
+      # So only the chia seeds are cancelled. The beans still need replenishing.
       expect { subject.amend_backorder(order) }
-        .to change { backorder.lines.count }.from(2).to(0)
+        .to change { backorder.lines.count }.from(2).to(1)
+        .and change { beans.reload.on_hand }.by(-12)
```

Comment:
I don't get where the -12 is coming from here, I am probably missing something. This is my understanding so far :
- we had -1 beans on hand, after amending backorder it becomes 11 as we back order another case of 12 can 
- order cancelled -> now we have 17 beans on hand as we return the 6 beans ordered
- after amending backorder, I would expect to get 17 - ( 2x12 ) = -7 beans on hand, be we have -12 ??

Reply:
No, it's not `to(-12)`, the change is `-12`. That is the difference. So we have 17 and take 12 off which would result in 5. I could have specified that here. We don't want to have negative stock locally, even after cancellations. We need to have positive stock to fulfill other placed orders, possibly in other order cycles.

Reply:
Thanks ! I knew I was missing something !